tests/c/testsuite/pwrite-with-access.c

```diff
@@ -0,0 +1,48 @@
+#include <assert.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+int main(int argc, char **argv) {
+  int ret;
+  int fd;
+  struct stat buf;
+  char *test_file = "fs-tests.dir/writeable/test_pwrite_pread.txt.cleanup";
+  char *full_content = "very long text";
+  const char *sub_content = "test";
+  char read_buf[16];
+
+  /* make sure the test file doesn't exist */
+  ret = access(test_file, F_OK);
+  assert(ret != 0);
+
+  fd = open(test_file, O_WRONLY | O_CREAT);
+  assert(fd > 0);
+
+  ret = pwrite(fd, full_content, strlen(full_content), 0);
+  assert(ret == strlen(full_content));
```

Comment:
This assert and the others like it are stricter than what POSIX requires, and I believe WASI wants to follow POSIX here. In particular, `pwrite` is permitted to write fewer bytes than requested, and still succeed.

Reply:
good point, updated the code.